### PR TITLE
Issue 16684 vanity url forward to url map pattern results in 404

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 1]
```diff
@@ -6,7 +6,10 @@
 import com.dotcms.rendering.velocity.viewtools.VelocityRequestWrapper;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 2]
```diff
@@ -6,7 +6,10 @@
 import com.dotcms.rendering.velocity.viewtools.VelocityRequestWrapper;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java [hunk 3]
```diff
@@ -286,6 +286,7 @@ private String buildContentQuery(
             query.append(this.getHostFilter(context.getHost()));
         }
 
+        query.append(" ");
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 4]
```diff
@@ -62,10 +65,19 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
...
+                        PageContextBuilder.builder()
+                                .setPageUri(uri)
+                                .setPageMode(mode)
+                                .setUser(APILocator.getLoginServiceAPI().getLoggedInUser())
+                                .build(),
+                        request,
+                        response
+                );
+
+                response.getOutputStream().write(pageHtml.getBytes());
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: here I am using the API to render a Page this internally do a "response.getOutputStream().write(pageHtml.getBytes())", but doing first several check as the vanity URL

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java [hunk 5]
```diff
@@ -280,6 +280,7 @@ private String buildContentQuery(
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

**Comment by fmontes** [comment]
This might be also fix issue#17594

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java [hunk 6]
```diff
@@ -280,6 +279,7 @@ private String buildContentQuery(
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java [hunk 7]
```diff
@@ -530,4 +532,14 @@ public String getQueryStringFromUri(final String uri) {
 				uri.substring(indexOf+1):
 				null;
 	}
+
+	public static String getCurrentURI(final HttpServletRequest request)  {
+		try {
+			return URLDecoder.decode((request.getAttribute(Constants.CMS_FILTER_URI_OVERRIDE) != null)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'Constants.CMS_FILTER_URI_OVERRIDE' due to existing static import 'com.dotmarketing.filters.Constants.CMS_FILTER_URI_OVERRIDE'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java [hunk 8]
```diff
@@ -68,12 +89,26 @@ public final String eval() {
...
     
-    public static final VelocityModeHandler modeHandler(PageMode mode, HttpServletRequest request, HttpServletResponse response, String uri, Host host) {
-        return pageModeVelocityMap.get(mode).apply(request, response, uri, host);
+    public static final VelocityModeHandler modeHandler(final PageMode mode, final HttpServletRequest request, final HttpServletResponse response, final String uri, final Host host) {
+        // Find the current language
+        final long langId = WebAPILocator.getLanguageWebAPI().getLanguage(request).getId();
+
+        try {
+            // now we check identifier cache first (which DOES NOT have a 404 cache )
+            final Identifier id = APILocator.getIdentifierAPI().find(host, uri);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: Codacy Issue found: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java [hunk 9]
```diff
@@ -74,7 +61,7 @@ public final void serve(final OutputStream out) throws DotDataException, IOExcep
 
 
             // now we check identifier cache first (which DOES NOT have a 404 cache )
-            Identifier id = APILocator.getIdentifierAPI().find(host, uri);
+            Identifier id = APILocator.getIdentifierAPI().find(this.htmlPage.getIdentifier());
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: Codacy Issue found: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 10]
```diff
@@ -6,7 +6,11 @@
 import com.dotcms.rendering.velocity.viewtools.VelocityRequestWrapper;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java [hunk 11]
```diff
@@ -42,6 +49,21 @@
...
+            final HttpServletResponse response,
+            final IHTMLPage htmlPage,
+            final Host host) {
+        this.request = request;
+        this.response = response;
+        this.htmlPage = htmlPage;
+        this.host = host;
+    }
+
+    public void setMode(PageMode mode) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: Codacy Issue found: [Parameter 'mode' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: Codacy Issue found: [Local variable 'id' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 12]
```diff
@@ -6,7 +6,11 @@
...
+import com.dotmarketing.exception.DotSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java [hunk 13]
```diff
@@ -530,4 +532,14 @@ public String getQueryStringFromUri(final String uri) {
...
+					? (String) request.getAttribute(Constants.CMS_FILTER_URI_OVERRIDE)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'Constants.CMS_FILTER_URI_OVERRIDE' due to existing static import 'com.dotmarketing.filters.Constants.CMS_FILTER_URI_OVERRIDE'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: Codacy Issue found: [Local variable 'id' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'Constants.CMS_FILTER_URI_OVERRIDE' due to existing static import 'com.dotmarketing.filters.Constants.CMS_FILTER_URI_OVERRIDE'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: Codacy Issue found: [Parameter 'mode' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: Codacy Issue found: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: Codacy Issue found: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'Constants.CMS_FILTER_URI_OVERRIDE' due to existing static import 'com.dotmarketing.filters.Constants.CMS_FILTER_URI_OVERRIDE'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: Codacy Issue found: [Local variable 'id' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'Constants.CMS_FILTER_URI_OVERRIDE' due to existing static import 'com.dotmarketing.filters.Constants.CMS_FILTER_URI_OVERRIDE'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java [hunk 14]
```diff
@@ -67,13 +88,35 @@ public final String eval() {
...
-    public static final VelocityModeHandler modeHandler(PageMode mode, HttpServletRequest request, HttpServletResponse response, String uri, Host host) {
-        return pageModeVelocityMap.get(mode).apply(request, response, uri, host);
+
+    public static final VelocityModeHandler modeHandler(final PageMode mode, final HttpServletRequest request, final HttpServletResponse response, final String uri, final Host host) {
+        // Find the current language
+        final long langId = WebAPILocator.getLanguageWebAPI().getLanguage(request).getId();
+
+        try {
+            // now we check identifier cache first (which DOES NOT have a 404 cache )
+            final Identifier id = APILocator.getIdentifierAPI().find(host, uri);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: Codacy Issue found: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: Codacy Issue found: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java]: Codacy Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java [hunk 15]
```diff
@@ -342,25 +367,33 @@ private IHTMLPage getPageByUri(final PageMode mode, final Host host, final Strin
         return htmlPage;
     }
 
+    private IHTMLPage getPageById(final PageMode mode, final Host host, final String id)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java]: Codacy Issue found: [Avoid unused method parameters such as 'host'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'Constants.CMS_FILTER_URI_OVERRIDE' due to existing static import 'com.dotmarketing.filters.Constants.CMS_FILTER_URI_OVERRIDE'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: Codacy Issue found: [Parameter 'mode' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java [hunk 16]
```diff
@@ -232,35 +245,43 @@ public String getPageHtml(
...
+            throw new HTMLPageAssetNotFoundException(context.getPageUri());
         }
 
-        htmlPageUrl.setHTMLPage((HTMLPageAsset) htmlPageAsset);
+        checkPagePermission(context, htmlPageUrl.htmlPage);
 
+        return htmlPageUrl;
+    }
+
+    private void checkPagePermission(PageContext context, IHTMLPage htmlPageAsset) throws DotDataException, DotSecurityException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java]: Codacy Issue found: [Parameter 'htmlPageAsset' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java (hunk 15 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java]: Codacy Issue found: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)

**Pull request closed without merging**